experimental/src/main/scala/org/testcontainers/testcontainers4s/lifecycle/TestLifecycleAware.scala

```diff
@@ -0,0 +1,11 @@
+package org.testcontainers.testcontainers4s.lifecycle
+
+import org.testcontainers.lifecycle.TestDescription
+import org.testcontainers.testcontainers4s.containers.Container
+
+trait TestLifecycleAware { self: Container =>
```

Comment:
Why do you need to re-introduce this interface, given that there is one already in the Java version of TC?

Reply:
Because it uses Java's `Optional`.

Reply:
I can't see it as a good motivation to introduce a duplicating interface.
testcontainers-scala should wrap testcontainers-java, not re-implement it.